taiga2/controllers/endpoint.py

```diff
@@ -1130,3 +1130,148 @@ def remove_group_user_associations(groupId, groupUserAssociationMetadata):
         return flask.jsonify(result)
     except AssertionError:
         return flask.abort(403)
+
+
+@validate
+def get_figshare_auth_url():
+    config = flask.current_app.config
+    return flask.jsonify(
+        {
+            "figshare_auth_url": figshare.AUTH_URL.format(
+                "?client_id={}".format(
+                    (config["FIGSHARE_CLIENT_ID"])
+                    + "&response_type=code"
+                    + "&scope=all"
+                    + "&redirect_uri%3Dhttps%3A%2F%2Fcds.team%2Ftaiga"
```

Comment:
does this url need to be hardcoded? Doesn't this also imply that this can't be tested locally?

Reply:
Yeah. The Figshare app needs a redirect URL, and if it doesn't match here, the request won't go through. Once it's deployed, we won't be able to test this as-is. One workaround could be creating a separate Figshare app for local tests that uses a different redirect url. Not ideal, but I'm not sure what else we could do :/